vm/src/frame.rs

```diff
@@ -176,32 +177,16 @@ impl Frame {
                     // 1. Extract traceback from exception's '__traceback__' attr.
                     // 2. Add new entry with current execution position (filename, lineno, code_object) to traceback.
                     // 3. Unwind block stack till appropriate handler is found.
-                    assert!(objtype::isinstance(
-                        &exception,
-                        &vm.ctx.exceptions.base_exception_type
-                    ));
 
-                    let traceback = vm
-                        .get_attribute(exception.clone(), "__traceback__")
-                        .unwrap();
-
-                    let next = if vm.is_none(&traceback) {
-                        None
-                    } else {
-                        let traceback: PyTracebackRef = traceback
-                            .downcast()
-                            .expect("next must be a traceback object");
-                        Some(traceback)
-                    };
+                    let next = exception.traceback();
 
                     let new_traceback = PyTraceback::new(
                         next,
                         self.clone().into_ref(vm),
                         self.lasti.get(),
                         lineno.row(),
                     );
-                    vm.set_attr(&exception, "__traceback__", new_traceback.into_ref(vm))
-                        .unwrap();
+                    exception.set_traceback(Some(new_traceback.into_ref(vm)));
```

Comment:
Does this also work when a user defined exception implements a custom setter function for the `__traceback__` member?

Reply:
It doesn't need to; CPython doesn't heed custom `__traceback__` setters either.